index.js

```diff
@@ -141,20 +141,38 @@ NYC.prototype.addAllFiles = function () {
   this.writeCoverageFile()
 }
 
+var hashCache = {}
+
+NYC.prototype._addSource = function (code, filename, relFile) {
```

Comment:
If you rename this to `_maybeInstrumentSource()` it's easier to see why it may return `null`. Then you could assign to a `instrumentedSource` variable and the logic in `addFile()` becomes more clear.